tf/tfprocess.py

```diff
@@ -268,6 +269,14 @@ def process(self, batch_size, test_batches, batch_splits=1):
             required_factor = 64 * self.cfg['training'].get('num_batch_splits', 1)
             raise ValueError('batch_size must be a multiple of {}'.format(required_factor))
 
+        # Determine learning rate
+        lr_values = self.cfg['training']['lr_values']
+        lr_boundaries = self.cfg['training']['lr_boundaries']
+        steps_total = steps % self.cfg['training']['total_steps']
+        self.lr = lr_values[bisect.bisect_right(lr_boundaries, steps_total)]
+        if self.warmup_steps > 0 and steps < self.warmup_steps:
+             self.lr = self.lr * steps / self.warmup_steps
```

Comment:
compute_update_ratio actually makes training crash because it doesn't like lr=0 at step=0 (all updates are 0 and are filtered away, then min of empty list is taken which crashes). `self.lr *= (steps + 1) / self.warmup_steps` should work.